src/FastRouteRouter.php

```diff
@@ -142,7 +142,7 @@ public function generateUri($name, array $substitutions = [])
         $path  = $route->getPath();
 
         foreach ($substitutions as $key => $value) {
-            $pattern = sprintf('#\{%s(:[^}]+)?\}#', preg_quote($key));
+            $pattern = sprintf('#\{%s(:[^}]+)?\}}#', preg_quote($key));
```

Comment:
This is problematic, as it assumes all placeholders use `{}` quantifiers internally; it's literally matching for `}}` as the closing delimiter, when some will only be `}`.